### PR TITLE
chore(doc): add weight description for Stage page

### PR DESCRIPTION
**Pull request by caozhuozi**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:

There is no description of the `weight` parameter on the Stage page.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

Please feel free to give comments and suggestions!

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by caozhuozi** [comment]
I'm also going to change the doc of `weight` in the API in this PR but haven't figured out how many files need to be changed and which script needs to be executed to generate the API doc.

```go
// Weight means the current stage, in case of multiple stages,
// a random stage will be matched as the next stage based on the weight.
Weight int
```
I think there may be a grammatical error in the current description.

**Comment by wzshiming** [comment]
> I'm also going to change the doc of `weight` in the API in this PR but haven't figured out how many files need to be changed and which script needs to be executed to generate the API doc.
> 
> ```go
> // Weight means the current stage, in case of multiple stages,
> // a random stage will be matched as the next stage based on the weight.
> Weight int
> ```
> 
> I think there may be a grammatical error in the current description.

Using `./hack/update-api-docs.sh` to do this, BTY `./hack/update-all.sh` can maintain all generated.

File: site/content/en/docs/user/stages-configuration.md [hunk 1]
```diff
@@ -61,6 +62,15 @@ and the changes that will be made to the resource when the stage is applied.
 The `next` field allows users to define the new state of the resource using the `statusTemplate` field,
 modify the `finalizers` of the resource, and even `delete` the resource.
 
+It is worth noting that there is no dedicated field for arranging the execution order if multiple stages of a resource type are provided.
+The execution order of stages can be naturally controlled by utilizing `selector.matchExpressions` and `next` field.
+Specifically, users can chain the stages by making `selector.matchExpressions` of a stage match the status content specified in `next` field of the previous stage.
```
Comment (windsonsea) [inline site/content/en/docs/user/stages-configuration.md:67]:
```suggestion
Specifically, users can chain the stages by ensuring that `selector.matchExpressions` of a stage match the status content specified in the `next` field of the previous stage.
```

File: site/content/en/docs/user/stages-configuration.md [hunk 2]
```diff
@@ -61,6 +62,15 @@ and the changes that will be made to the resource when the stage is applied.
 The `next` field allows users to define the new state of the resource using the `statusTemplate` field,
 modify the `finalizers` of the resource, and even `delete` the resource.
 
+It is worth noting that there is no dedicated field for arranging the execution order if multiple stages of a resource type are provided.
+The execution order of stages can be naturally controlled by utilizing `selector.matchExpressions` and `next` field.
```
Comment (windsonsea) [inline site/content/en/docs/user/stages-configuration.md:66]:
```suggestion
The execution order of stages can be controlled by utilizing `selector.matchExpressions` and `next` field.
```
`naturally` may cause an intention to ask what is natural. Could we remove it?

**Review by windsonsea — COMMENTED** [review]
Some text suggestions for reference.

File: site/content/en/docs/user/stages-configuration.md (hunk 2 above)
Comment (caozhuozi) [reply site/content/en/docs/user/stages-configuration.md]: "naturally" is not necessary here. Thanks for improving the writing!

**Review by windsonsea — COMMENTED** [review]
/lgtm

Thanks for adding this weight desc for stages

**Comment by caozhuozi** [comment]
> > I'm also going to change the doc of `weight` in the API in this PR but haven't figured out how many files need to be changed and which script needs to be executed to generate the API doc.
> > ```go
> > // Weight means the current stage, in case of multiple stages,
> > // a random stage will be matched as the next stage based on the weight.
> > Weight int
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I think there may be a grammatical error in the current description.
> 
> Using `./hack/update-api-docs.sh` to do this, BTY `./hack/update-all.sh` can maintain all generated.

Thanks. I fixed the grammatical errors with some simple descriptions.

**Comment by wzshiming** [comment]
It also needs `./hack/update-controllergen.sh` to update the CRDs.

**Comment by caozhuozi** [comment]
> It also needs `./hack/update-controllergen.sh` to update the CRDs.

Thanks! There is still a lot to learn:)

**Pull request merged**